diff --git a/orig.spec b/nginx.spec
index 1511d7f..913cfe6 100644
--- a/orig.spec
+++ b/nginx.spec
@@ -22,6 +22,8 @@ BuildRequires: redhat-lsb-core
 %define epoch 1
 Epoch: %{epoch}
 Requires(pre): shadow-utils
+Requires: luajit
+Requires: openldap
 Requires: systemd
 BuildRequires: systemd
 %define os_minor %(lsb_release -rs | cut -d '.' -f 2)
@@ -78,6 +80,8 @@ Source10: nginx.suse.logrotate
 Source11: nginx-debug.service
 Source12: COPYRIGHT
 Source13: nginx.check-reload.sh
+Source14: nginx-auth-ldap-master.tar.gz
+Source15: lua-nginx-module-master.tar.gz
 
 License: 2-clause BSD-like license
 
@@ -96,7 +100,7 @@ a mail proxy server.
 %endif
 
 %prep
-%setup -q
+%setup -a 14 -a 15 -q
 cp %{SOURCE2} .
 sed -e 's|%%DEFAULTSTART%%|2 3 4 5|g' -e 's|%%DEFAULTSTOP%%|0 1 6|g' \
     -e 's|%%PROVIDES%%|nginx|g' < %{SOURCE2} > nginx.init
@@ -107,13 +111,17 @@ sed -e 's|%%DEFAULTSTART%%||g' -e 's|%%DEFAULTSTOP%%|0 1 2 3 4 5 6|g' \
 ./configure %{BASE_CONFIGURE_ARGS} \
     --with-cc-opt="%{WITH_CC_OPT}" \
     --with-ld-opt="%{WITH_LD_OPT}" \
-    --with-debug
+    --with-debug \
+    --add-module=nginx-auth-ldap-master \
+    --add-module=lua-nginx-module-master
 make %{?_smp_mflags}
 %{__mv} %{bdir}/objs/nginx \
     %{bdir}/objs/nginx-debug
 ./configure %{BASE_CONFIGURE_ARGS} \
     --with-cc-opt="%{WITH_CC_OPT}" \
-    --with-ld-opt="%{WITH_LD_OPT}"
+    --with-ld-opt="%{WITH_LD_OPT}" \
+    --add-module=nginx-auth-ldap-master \
+    --add-module=lua-nginx-module-master
 make %{?_smp_mflags}
 
 %install
